src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -90,6 +91,22 @@ public Recipe modifyRecipeStep(int stepNumber, int newStepNumber) {
         return new Recipe(this.uuid.getId(), this.name, this.ingredientList, stepCopy);
     }
 
+    /**
+     * Changes the specified ingredient with a new ingredient.
+     */
+    public Recipe modifyIngredients(String oldIngredient, Ingredient newIngredient) {
+        requireAllNonNull(oldIngredient, newIngredient);
+        for (Ingredient ingredient : ingredientList) {
+            if (ingredient.getName().fullName.equals(oldIngredient)) {
```

Comment:
I think this part can make the correct path more clear based on the code quality standard